### PR TITLE
cmd/snap: make completion skip hidden commands (unless overridden)

### PR DESCRIPTION
**Pull request by pedronis**
This stops completion including all hidden commands. It allow this to be overridden via cmdInfo.completeHidden for the few commands that are hidden but generally used (we have a few related to keys and signing).

### CODE REVIEW

**Comment by pedronis** [comment]
Built on top of #7688

File: cmd/snap/main.go [hunk 1]
```diff
@@ -88,10 +88,13 @@ type cmdInfo struct {
 	name, shortHelp, longHelp string
 	builder                   func() flags.Commander
 	hidden                    bool
-	optDescs                  map[string]string
-	argDescs                  []argDesc
-	alias                     string
-	extra                     func(*flags.Command)
+	// autocompletHidden set to true forces autocompletion even of
```
Comment (chipaca) [inline cmd/snap/main.go]: it's completion, not autocompletion, so `completeHidden` would be better :-)

**Comment by pedronis** [comment]
@chipaca this breaks tests/main/completion, I don't have time myself this week but please dig when you do

File: cmd/snap/main.go [hunk 2]
```diff
@@ -211,6 +211,9 @@ var noCompletion = make(map[string]bool)
 
 func markForNoCompletion(ci *cmdInfo) {
 	if ci.hidden && !ci.completeHidden {
+		if ci.shortHelp == "" {
+			logger.Panicf("%d missing short help", ci.name)
```
Comment (pedronis) [inline cmd/snap/main.go]: %d ?

Comment (chipaca) [reply cmd/snap/main.go]: d'oh

Comment (chipaca) [reply cmd/snap/main.go]:
```suggestion
			logger.Panicf("%q missing short help", ci.name)
```

**Pull request merged**